var/spack/repos/builtin/packages/py-bglibpy/package.py

```diff
@@ -27,8 +27,10 @@ class PyBglibpy(PythonPackage):
     version('4.1.4', commit='e54d294460e5bdf6b9990bc10a4606b412b76d90')
 
     depends_on('py-setuptools', type=('build', 'run'))
-
     depends_on('neuron+python', type='run')
+    depends_on('py-numpy', type='run')
+    depends_on('py-matplotlib', type='run')
+    depends_on('py-cachetools', type='run')
     depends_on('py-h5py~mpi@2.3:', type='run')
     depends_on('py-bluepy', type='run')
```

Comment:
`py-bglibpy` 4.4.36 should depend on `py-bluepy` >= 2.1, as in `setup.py`.
Shouldn't the version numbers be added to the dependencies?

Reply:
Thanks @GianlucaFicarelli, good point.
`class PyBglibpy(PythonPackage)` has multiple versions on it.
When I specify:
`depends_on('py-bluepy@2.1', type='run')`
Is it only going to affect the latest version or is it going to effect the older versions as well?

Reply:
If you need to keep the older versions you can add a `when` clause to specify that the dependency depends on the version.
This is an example (maybe not completely correct), but you could also take a look at `py-bluepy/package.py` to see how complicated it can become :D

```
    depends_on('py-bluepy@:1.999', type='run', when='@:4.4.35')
    depends_on('py-bluepy@2.1:', type='run', , when='@4.4.36:')
```

Reply:
I see, thanks! It can get complicated indeed :D